crates/vm/src/expr.rs

```diff
@@ -713,44 +728,44 @@ impl QueryExpr {
                     },
                 ..
             }) if table.table_id != rhs_table_id => {
-                self = self.with_index_lower_bound(table, col_id, value, inclusive);
+                self = self.with_index_lower_bound(table, columns, value, inclusive);
                 self.query.push(query);
                 self
             }
             // try to push below join's rhs
             Query::JoinInner(JoinExpr { rhs, col_lhs, col_rhs }) => {
                 self.query.push(Query::JoinInner(JoinExpr {
-                    rhs: rhs.with_index_lower_bound(table, col_id, value, inclusive),
+                    rhs: rhs.with_index_lower_bound(table, columns, value, inclusive),
                     col_lhs,
                     col_rhs,
                 }));
                 self
             }
             // merge with a preceding upper bounded index scan (inclusive)
             Query::IndexScan(IndexScan {
-                col_id: lhs_col_id,
+                columns: lhs_col_id,
```

Comment:
I'm guessing that future work will extend index scans to multiple columns?

Reply:
Yes, this come in the next PR.